### ISSUE TITLE
Threshholding for output probabilities

### ISSUE THREAD

**Issue by anupamme** [aligholami/CheXpert-Keras#3]
Hello 

I am looking into test.py and you are using roc_auc_score from sklearn metrics. I could not find this metric on the chexpert paper. Why the choice of this metric?

For my use case, I need to predicted output as 0 or 1 (instead of probabilities). So I need threshold values for each category above which output should be 1 and below it should be 0. Can you suggest how to achieve this in this codebase?

**Comment by aligholami** [comment]
I choose this metric since its a common metric in the community. It is strong against variations that may occur in other metrics such as Accuracy (Jaccard distance), Precision and Recall. This measure is called Area Under the ROC Curve and you may find more info here:
https://en.wikipedia.org/wiki/Receiver_operating_characteristic

**Comment by aligholami** [comment]
The default test model is the model you used for training. Going back to the train.py file and you see the loss function **Binary Cross Entropy** in the model compilation line:

` 
model_train.compile(optimizer=optimizer, loss="binary_crossentropy")
`

this line says that the model's predictions, which are a bunch of numbers between 0 and 1 (because of a Sigmoid that is applied on predictions inside this loss function by default). So, in order to get predictions as you wish, between 0 and 1, you need to grab the predictions of the model, apply a Sigmoid on them (to get numbers between 0 and 1), and pass them through whatever loss or any other function you may want.

On how to get predictions:
https://stackoverflow.com/questions/45587378/how-to-get-predicted-values-in-keras

**Comment by aligholami** [comment]
If you just want either 0 or 1, you should apply a **round** function to round the numbers to either 0 or 1.

**Comment by anupamme** [comment]
Thanks for the response. I had a followup question:

The paper uses 3-class classification as well (page 4), I am quoting from it:

"In this approach (U-MultiClass model), for each observation, we output the probability of each of the 3 possibleclasses{p0,p1,pu}∈[0,1],p0+p1+pu =1.We set up the loss as the mean of the multi-class cross-entropy losses over the observations. At test time, for the probability of a particular observation, we output the probability of the positive label after applying a softmax restricted to the positive and negative classes."

Did you manage to implement 3 class classification as well?
